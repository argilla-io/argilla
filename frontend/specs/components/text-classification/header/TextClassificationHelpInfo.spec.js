import { shallowMount } from "@vue/test-utils";
import ComponentTextClassifierHelpInfo from "@/components/text-classifier/header/TextClassificationHelpInfo";

const props = {
  visible: true,
};

let wrapper = null;
beforeEach(() => {
  wrapper = shallowMount(ComponentTextClassifierHelpInfo, {
    stubs: ["lazy-base-modal", "base-button"],
    propsData: props,
  });
});
afterEach(() => {
  wrapper.destroy();
});
describe("TextClassificationHelpInfo", () => {
  let spy = jest.spyOn(console, "error");
  afterEach(() => spy.mockReset());

  test("renders properly", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
