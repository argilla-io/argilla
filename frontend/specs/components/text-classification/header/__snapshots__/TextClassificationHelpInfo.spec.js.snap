// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextClassificationhelpInfo renders properly 1`] = `
<lazy-base-modal modal-class="modal-secondary" modal-position="modal-top-right" modal-title="Info" modal-icon="info" modal-custom="true" modal-visible="true">
  <div class="help-info__content">
    <p class="help-info__title">Highlight colors in token attributions</p>
    <p>
      Argilla enables you to register token attributions as part of the
      dataset records. For getting token attributions, you can use methods
      such as Integrated Gradients or SHAP. These methods try to provide a
      mechanism to interpret model predictions.
    </p>
    <p>The attributions work as follows:</p>
    <p>
      [0,1] <strong>Positive attributions</strong> (in blue) reflect those
      tokens that are making the model predict the specific predicted label.
    </p>
    <p>
      [-1, 0] <strong>Negative attributions</strong> (in red) reflect those
      tokens that can influence the model to predict a label other than the
      specific predicted label.
    </p>
  </div>
  <div class="modal-buttons">
    <base-button class="primary">Ok, got it!</base-button>
    <base-button href="https://docs.rubrix.ml/en/stable/tutorials/nlp_model_explainability.html" target="_blank" class="primary link">More in docs</base-button>
  </div>
</lazy-base-modal>
`;
